drivers/led_strip/ws2812_spi.c

```diff
@@ -86,24 +86,27 @@ static int ws2812_strip_update_rgb(const struct device *dev,
 	const uint8_t one = cfg->one_frame, zero = cfg->zero_frame;
 	struct spi_buf buf = {
 		.buf = cfg->px_buf,
-#if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_PREAMBLE_SUPPORT)
-		.len = cfg->empty_preamble_frames + (cfg->length * 8 * cfg->num_colors),
-#else
 		.len = (cfg->length * 8 * cfg->num_colors),
-#endif
 	};
 	const struct spi_buf_set tx = {
 		.buffers = &buf,
 		.count = 1
 	};
 	size_t i;
 	int rc;
+#if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_POSTAMBLE_SUPPORT)
+    buf.len += cfg->empty_padding_frames;
+#endif
 #if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_PREAMBLE_SUPPORT)
-	uint8_t *px_buf = &cfg->px_buf[cfg->empty_preamble_frames];
+    buf.len += cfg->empty_padding_frames;
+	uint8_t *px_buf = &cfg->px_buf[cfg->empty_padding_frames];
 	/* Ensure zero fill on preamble (if present) */
-	memset(cfg->px_buf, 0, cfg->empty_preamble_frames);
 #else
 	uint8_t *px_buf = cfg->px_buf;
+#endif
+#if IS_ENABLED(CONFIG_WS2812_STRIP_SPI_PREAMBLE_SUPPORT) || IS_ENABLED(CONFIG_WS2812_STRIP_SPI_POSTAMBLE_SUPPORT)
+    /* If zero values are padded on either end we need to ensure the buffer is all zeros before filling */
+	memset(cfg->px_buf, 0, buf.len);
```

Comment:
As it stands it appears the buffer is not initialised, so it could and likely does have non zero values before this statement. That being said, it might be better to zero init the buffer during declaration on line 235